### PR TITLE
Several service fixes

### LINKED ISSUE #121: Use crt.sh SQL access instead of web scrapping
Hi,

I noticed you use web scrapping to retrieve results from crt.sh, but there is a better way (web interface may timeout on long requests): use the SQL access https://groups.google.com/forum/#!topic/crtsh/oEDOzwr2Fuc

Here is a PoC in Go:
https://github.com/z0mbie42/crtsh/blob/master/main.go

The weird thing with 
```go
idx := strings.Index(subdomainsPattern, "%")
idxrev := strings.Index(reverseStr(subdomainsPattern), "%")
```
is to greatly speedup query because it's a wildcard match.

Best regards.

### PR DESCRIPTION
**Pull request by caffix**
#121 #139

### CODE REVIEW

File: amass/amass.go [hunk 1]
```diff
@@ -247,10 +222,39 @@ loop:
...
+func (e *Enumeration) periodicChecks(services []core.Service) {
+	done := true
+	for _, srv := range services {
+		if srv.IsActive() {
+			done = false
+			break
+		}
+	}
+	if done {
+		close(e.Done)
```
Comment (kalbasit) [inline amass/amass.go]: So if any of the services are still active, you don't ever come back and close `e.Done`? Did you intend to have a loop over this?

File: amass/sources/sources.go [hunk 2]
```diff
@@ -20,15 +20,22 @@ import (
...
-	}
 	name = strings.TrimSpace(strings.ToLower(name))
+
+	for {
+		if i := nameStripRE.FindStringIndex(name); i != nil {
+			name = name[i[1]:]
+		} else {
+			break
+		}
+	}
```
Comment (kalbasit) [inline amass/sources/sources.go]: This seems pretty complex function. Do you have a unit test covering it?

File: amass/amass.go (hunk 1 above)
Comment (caffix) [reply amass/amass.go]: If any of the services are still active, e.Done is not closed, which will prevent the loop from terminating. Once all the services are inactive, then the loop will terminate

File: amass/sources/sources.go (hunk 2 above)
Comment (caffix) [reply amass/sources/sources.go]: You're right. There's no unit test for it yet, so we'll need to add one. For now though, these fixes need to be made available to users

File: amass/amass.go (hunk 1 above)
Comment (kalbasit) [reply amass/amass.go]: What loop are you referring to? The range loop will stop once it runs out of elements in the services array.

Comment (caffix) [reply amass/amass.go]: The loop at the end of the Start() method that calls the periodicChecks method

Comment (kalbasit) [reply amass/amass.go]: Oh, I see now. I should have looked at the file as a whole :smile:.

**Pull request merged**